lib/OpenQA/CacheService/Model/Cache.pm

```diff
@@ -352,18 +297,20 @@ sub _check_limits {
     my $limit = $self->limit;
     my $log   = $self->log;
 
-    eval {
-        my $db      = $self->sqlite->db;
-        my $results = $db->select('assets', [qw(filename size last_use)], undef, {-asc => 'last_use'});
-        for my $asset ($results->hashes->each) {
-            my $asset_size = $asset->{size} || -s $asset->{filename};
-            $self->_decrease($asset_size) if $self->purge_asset($asset->{filename});
-            last if !$self->_exceeds_limit($needed);
-        }
-    } if $self->_exceeds_limit($needed);
-    if ($@) { $log->error("CACHE: check_limit failed: $@") }
-
-    $log->debug("CACHE: Health: Real size: $self->{cache_real_size}, Configured limit: $limit");
+    if ($self->_exceeds_limit($needed)) {
+        $log->info("Cache size $self->{cache_real_size} + $needed exceeds limit of $limit, purging least used assets");
+        eval {
+            my $db      = $self->sqlite->db;
+            my $results = $db->select('assets', [qw(filename size last_use)], undef, {-asc => 'last_use'});
+            for my $asset ($results->hashes->each) {
+                my $asset_size = $asset->{size} || -s $asset->{filename} || 0;
+                $log->info("Purging $asset->{filename} because we need space for new assets, reclaiming $asset_size");
+                $self->_decrease($asset_size) if $self->purge_asset($asset->{filename});
+                last if !$self->_exceeds_limit($needed);
+            }
+        };
+        if ($@) { $log->error("Checking cache limit failed: $@") }
```

Comment:
I thought normally you would also prefer post-conditions, e.g.

```perl
$log->error("Checking cache limit failed: $@") if $@;
```

Reply:
Not when `eval` is involved, that's a special case to me (`try/catch`) where i always use blocks.

Reply:
ok, got it. Need to learn this style then ;)

Reply:
I like to use this form usually:
```perl
eval {
  ...
};
if (my $err = $@) {
  ...
}
```
Assigning `$@` to a variable as soon as possible, to avoid any chance of having it accidentally re-assigned to something else by something in the exception handling code. (I will probably change the code above now to use `$err`, just to be more consistent)